ddtrace/contrib/tornado/decorators.py

```diff
@@ -0,0 +1,45 @@
+from tornado.concurrent import Future
+
+from .settings import FUTURE_SPAN_KEY
+
+
+def _finish_coroutine_span(future):
+    """
+    Finish the opened span if it's attached to the given ``Future``
+    object. This method is a Tornado callback, that is used to close
+    a decorated coroutine.
+    """
+    span = getattr(future, FUTURE_SPAN_KEY, None)
+    if span:
+        # retrieve the exception info from the Future (if any)
+        exc_info = future.exc_info()
+        if exc_info:
+            span.set_exc_info(*exc_info)
+
+        span.finish()
+
+def wrap_executor(tracer, fn, args, kwargs, span_name, service, resource, span_type):
```

Comment:
as a follow-up of https://github.com/DataDog/dd-trace-py/pull/221#pullrequestreview-30028548 we should update the signature so that this is a valid example for future integrations.